tuning/sft_trainer.py

```diff
@@ -128,12 +134,16 @@ def train(
         raise ValueError("gradient_accumulation_steps has to be an integer >= 1")
 
     task_type = "CAUSAL_LM"
+    additional_metrics = {}
+
+    model_load_time = time.time()
     model = AutoModelForCausalLM.from_pretrained(
         model_args.model_name_or_path,
         cache_dir=train_args.cache_dir,
         torch_dtype=get_torch_dtype(model_args.torch_dtype),
         attn_implementation="flash_attention_2" if model_args.use_flash_attn else None,
     )
+    additional_metrics["model_load_time"] = time.time() - model_load_time
```

Comment:
wondering if this should also include tokenizer load time as its associated with model? should we move this block here `# TODO: Move these to a config as well
    tokenizer = AutoTokenizer.from_pretrained(
        model_args.model_name_or_path, cache_dir=train_args.cache_dir, use_fast=True
    )` 
    what do you think @dushyantbehl ?

Reply:
@Ssukriti model load time here just refers to the overhead of filesystem or COS bucket and was just added as an example on how to track extra metrics form the new api. 
I can certainly include tokenizer time or even have it separately.

Reply:
yes I understand its an example, but since you did the example maybe lets add tokenizer time in it too  for completeness :) . From what I see online model load time includes loading of model + associated tokenizer. Thanks

Reply:
@Ssukriti added and updated the code.